src/option.h

```diff
@@ -478,23 +478,74 @@ namespace librealsense
         std::function<void(const option&)> _recording_function = [](const option&) {};
     };
 
+    /** Wrapper for another option -- forwards all API calls to the proxied option
+    *such that specific functionality can be easily overriden */
+    class proxy_option : public option
+    {
+    public:
+        explicit proxy_option(std::shared_ptr<option> proxy_option)
+            : _proxy(proxy_option)
+        {}
+
+        const char* get_value_description(float val) const override
+        {
+            return _proxy->get_value_description(val);
+        }
+        const char* get_description() const override
+        {
+            return _proxy->get_description();
+        }
+        virtual void set(float value) override 
+        {
+            return _proxy->set(value);
+        }
+        float query() const override
+        {
+            return _proxy->query();
+        }
+
+        option_range get_range() const override
+        {
+            return _proxy->get_range();
+        }
+
+        bool is_enabled() const override
+        {
+            return  _proxy->is_enabled();
+        }
+
+        bool is_read_only() const override
+        {
+            return  _proxy->is_read_only();
+        }
+
+        void enable_recording(std::function<void(const option &)> record_action) override
+        {
+            _recording_function = record_action;
```

Comment:
Shouldn't this call `_proxy->enable_recording(...)`?

I'm OK leaving it as-is, since that's the way it was before, but I'll get @ev-mp input...